In [1]:
import pandas as pd
import numpy as np

In [2]:
data = pd.read_csv('visit-sales.csv')

In [3]:
data.head()

visit number  product 1  product 2  product 3
0             1      89.74      48.72      24.78
1             2        NaN      65.23      14.10
2             3      36.58      65.25      26.30
3             4      47.59        NaN      67.39
4             5      85.93      41.80      83.72

In [4]:
sample_2 = data['product 2']

In [5]:
sample_2.dropna(inplace = True)
sample_2.reset_index(inplace = True, drop = True)

In [6]:
sample_2.head()

0    48.72
1    65.23
2    65.25
3    41.80
4    23.70
Name: product 2, dtype: float64

In [7]:
sample_2.describe()

count    45.000000
mean     47.635333
std      27.983043
min       4.810000
25%      20.200000
50%      48.720000
75%      68.850000
max      94.250000
Name: product 2, dtype: float64

H0 = 30

HA > 30

Page mean = 47.64, is the sample mean, not the popularity mean

cannot reject the null hypothesis

In [8]:
import scipy.stats as stats

In [11]:
stats.ttest_1samp(sample_2, 30, alternative = 'greater')
#pvalue is very small, null hypothesis should be rejected

Ttest_1sampResult(statistic=4.227610670789385, pvalue=5.8684422616986934e-05)

In [12]:
xbar = sample_2.mean()
sem = stats.sem(sample_2)
degree = len(sample_2) - 1
stats.t.interval(0.95, degree, xbar, sem)
#population mean of Page B

(39.22829643890514, 56.042370227761516)

In [13]:
sample_1 = data['product 1']
sample_1.dropna(inplace = True)
sample_1.reset_index(inplace = True, drop = True)

In [15]:
stats.ttest_ind(sample_1, sample_2, equal_var = True, alternative = 'two-sided')
#pvalue is not super small, cannot reject the null hypothesis

Ttest_indResult(statistic=0.5748527583914115, pvalue=0.5668094101511096)

In [16]:
sample_3 = data['product 3']
sample_3.dropna(inplace = True)
sample_3.reset_index(inplace = True, drop = True)

In [17]:
stats.f_oneway(sample_1, sample_2, sample_3)

F_onewayResult(statistic=0.1714355642870647, pvalue=0.8426390311693355)